tests/sources/test_dropbox_aiohttp.py

```diff
@@ -0,0 +1,809 @@
+                source.dropbox_client.api_call(
+                    base_url=HOST_URLS["FILES_FOLDERS_HOST_URL"],
+                    url_name=PING,
+                    data=json.dumps(None),
+                )
+            )
+
+
+@pytest.mark.asyncio
+async def test_api_call():
+    source = create_source(DropboxDataSource)
+    source.dropbox_client._set_access_token = AsyncMock()
+
+    with patch.object(
+        aiohttp.ClientSession,
+        "post",
+        return_value=get_json_mock(MOCK_CURRENT_USER, 200),
+    ):
+        EXPECTED_RESPONSE = {
+            "account_id": "acc_id:1234",
+            "name": {
+                "given_name": "John",
+                "surname": "Wilber",
+                "display_name": "John Wilber",
+                "abbreviated_name": "JW",
+            },
+            "email": "john.wilber@abcd.com",
+            "country": "US",
+        }
+        response = await anext(
+            source.dropbox_client.api_call(
+                base_url=HOST_URLS["FILES_FOLDERS_HOST_URL"],
+                url_name=PING,
+                data=json.dumps(None),
+            )
+        )
+        actual_response = await response.json()
+        assert actual_response == EXPECTED_RESPONSE
+
+
+@pytest.mark.asyncio
+async def test_set_access_token_when_token_expires_at_is_str():
+    source = create_source(DropboxDataSource)
+    source.dropbox_client.token_expiration_time = "2023-02-10T09:02:23.629821"
+    mock_token = {"access_token": "test2344", "expires_in": "1234555"}
+    async_response_token = get_json_mock(mock_token, 200)
+
+    with patch.object(aiohttp.ClientSession, "post", return_value=async_response_token):
+        actual_response = await source.dropbox_client._set_access_token()
+        assert actual_response is None
+
+
+@pytest.fixture
+def patch_default_wait_multiplier():
+    with mock.patch("connectors.sources.dropbox.RETRY_INTERVAL", 0):
+        yield
+
+
+@pytest.mark.asyncio
+@mock.patch("connectors.utils.apply_retry_strategy")
+async def test_api_call_when_token_is_expired(
+    mock_apply_retry_strategy,
+):
+    source = create_source(DropboxDataSource)
+    mock_apply_retry_strategy.return_value = mock.Mock()
+
+    with patch.object(
+        aiohttp.ClientSession,
+        "post",
+        side_effect=ClientResponseError(
+            status=401,
+            request_info=aiohttp.RequestInfo(
+                real_url="", method=None, headers=None, url=""
+            ),
+            history=None,
+            message="Unauthorized",
+        ),
+    ):
+        with pytest.raises(ClientResponseError):
+            await anext(
+                source.dropbox_client.api_call(
+                    base_url=HOST_URLS["FILES_FOLDERS_HOST_URL"],
+                    url_name=PING,
+                    data=json.dumps(None),
+                )
+            )
+        await source.close()
+
+
+@pytest.mark.asyncio
+async def test_api_call_when_status_429_exception():
```

Comment:
Shouldn't this assert that a retry is made, not that it raises an exception?

Reply:
Yes, we can also do that. After retrying once, we can return an actual response 💡